lib/chef/provider/git.rb

```diff
@@ -355,6 +352,28 @@ def repo_url
           new_resource.repository
         end
       end
+
+      # Returns the home directory of the user
+      # @param [String] user must be a string.
+      # @return [String] the home directory of the user.
+      #
+      def get_homedir(user)
+        require "etc" unless defined?(Etc)
+
+        case user
+        when Integer
+          Etc.getpwuid(user).dir
+        else
+          Etc.getpwnam(user.to_s).dir
+        end
+      end
+
+      # Rescues the error if the user does not exist
+      def safe_get_homedir(user)
```

Comment:
We should probably just do this rescue in the assertion instead of adding this extra method. It just makes understanding what we're doing in the assertion a bit tricker.